src/plugins/intel_cpu/tests/unit/shape_inference_test/col2im_shape_inference_test.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "utils.hpp"
+
+using namespace ov;
+using namespace ov::intel_cpu;
+using ov::op::v0::Constant;
+using ov::op::v0::Parameter;
+using testing::HasSubstr;
+
+class Col2ImStaticTestSuite : public ::testing::TestWithParam<std::tuple<Shape,         // data shape
+                                                              std::vector<int32_t>,     // output_size values
+                                                              std::vector<int32_t>,     // kernel_size values
+                                                              Strides,                  // strides
+                                                              Strides,                  // dilations
+                                                              Shape,                    // pads_begin
+                                                              Shape,                    // pads_end
+                                                              Shape>> {};               // expected output shape
+
+TEST_P(Col2ImStaticTestSuite, Col2ImStaticShapeInference) {
+    const auto data_shape = std::get<0>(GetParam());
+    const auto output_size_val = std::get<1>(GetParam());
+    const auto kernel_size_val = std::get<2>(GetParam());
+    const auto strides = std::get<3>(GetParam());
+    const auto dilations = std::get<4>(GetParam());
+    const auto pads_begin = std::get<5>(GetParam());
+    const auto pads_end = std::get<6>(GetParam());
+    const auto expected_output_shape = std::get<7>(GetParam());
+
+    const auto data = std::make_shared<Parameter>(element::i64, data_shape);
+    const auto output_size = std::make_shared<op::v0::Constant>(element::i64, Shape{2}, output_size_val);
+    const auto kernel_size = std::make_shared<op::v0::Constant>(element::i64, Shape{2}, kernel_size_val);
+
+    const auto op = std::make_shared<op::v15::Col2Im>(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end);
+
+    const auto input_shapes = ShapeVector{data_shape, Shape{2}, Shape{2}};
+    auto shape_infer = make_shape_inference(op);
```

Comment:
This test still uses the fallback path (PartialShapes instead StaticShape) because the shape_infer is not added to shape inference factory and template function `shape_infer` is not build for static shape.

Consider update shape inference factory or add single test with default created op and call `shape_infer` directly to build it for static shape.